test/gtest/ucp/ucp_test.cc

```diff
@@ -286,14 +286,13 @@ ucs_status_t ucp_test::request_process(void *req, int worker_index, bool wait,
     ucs_status_t status;
     if (wait) {
         ucs_time_t deadline = ucs::get_deadline();
-        while (((status = ucp_request_check_status(req)) ==
-                UCS_INPROGRESS) && (ucs_get_time() < deadline)) {
+        while ((ucp_request_check_status(req) == UCS_INPROGRESS) &&
+               (ucs_get_time() < deadline)) {
```

Comment:
`if (wait)` can be part of `while()`, see my comment before

Reply:
done